core/operations/management/query_index_drop.cxx

```diff
@@ -18,38 +18,33 @@
 #include "query_index_drop.hxx"
 
 #include "core/utils/json.hxx"
+#include "core/utils/keyspace.hxx"
 #include "error_utils.hxx"
 
 namespace couchbase::core::operations::management
 {
 std::error_code
 query_index_drop_request::encode_to(encoded_request_type& encoded, http_context& /*context*/) const
 {
-    if ((scope_name.empty() && !collection_name.empty()) || (!scope_name.empty() && collection_name.empty())) {
+    if (!utils::check_query_management_request(*this)) {
         return errc::common::invalid_argument;
     }
     encoded.headers["content-type"] = "application/json";
-    std::string query_context = fmt::format("{}:`{}`", namespace_id, bucket_name);
-    auto keyspace = query_context;
-    if (!scope_name.empty()) {
-        query_context += ".`" + scope_name + "`";
-        keyspace += ".`" + scope_name + "`";
-    } else {
-        query_context += fmt::format(".`{}`", couchbase::scope::default_name);
-    }
-    if (!collection_name.empty()) {
-        keyspace += ".`" + collection_name + "`";
-    }
+    auto keyspace = utils::build_keyspace(*this);
     std::string drop_index_stmt;
     if (is_primary && index_name.empty()) {
-        drop_index_stmt = fmt::format(R"(DROP PRIMARY INDEX ON {} USING GSI)", keyspace);
+        drop_index_stmt = fmt::format(R"(DROP PRIMARY INDEX ON {})", keyspace);
```

Comment:
Should we keep the `USING GSI` portion of the query statement (same w/ other queries below)?

Reply:
Oh -- ya turns out that is really optional.   But...  I should keep it throughout.   I'll fix.